### PR TITLE
Issue #170 - Add folder based journal

### PR DESCRIPTION
**Pull request by notbalanced**
Adds feature for issue #170 (and #398) where you configure your journal to be a directory and entries are added as sub-directories and files: yyyy/mm/dd.txt. Multiple entries in a day will go in the same file, but a
new entry for a specific day will create a new file (and directory structure).

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -89,6 +89,7 @@ def get_version(filename="jrnl/__init__.py"):
         "keyring>=7.3",
         "passlib>=1.6.2",
         "pyxdg>=0.25",
+        "asteval>=0.9.8",
```
Comment (MinchinWeb) [inline setup.py]: I don't think this (`asteval`) is being used...

Comment (notbalanced) [inline setup.py]: Yeah, you're right. It isn't in my checkin, but if you look at this
[checkin](https://github.com/maebert/jrnl/commit/8b64cc95d9e5171b7b914451f8538783b77c2c71), it adds `asteval` in [template.py](https://github.com/maebert/jrnl/commit/8b64cc95d9e5171b7b914451f8538783b77c2c71#diff-46697bd48828cb658c9837ea175fab2fR2).

When the Travis build ran, it was failing, so I added it to fix that dependency.

I didn't know how to handle that. Is there a better way of doing that?

Comment (MinchinWeb) [inline setup.py]: No, this is probably a great way to handle it. Great catch.

(If anyone else is following up on this, `template.py` has been moved to [jrnl/plugins/template.py](https://github.com/maebert/jrnl/blob/2.0-rc1/jrnl/plugins/template.py#L2))

**Comment by wren** [comment]
This seems very useful!

**Comment by notbalanced** [comment]
I'd still like to see this merged in with jrnl.

**Comment by wren** [comment]
Hi, @notbalanced. I would also like to see this merged in. Do you have a chance to resolve the merge conflicts?

**Comment by notbalanced** [comment]
Merge conflicts have been resolved and I've updated the tests so that they pass correctly.

**Comment by wren** [comment]
Thank you, @notbalanced! I'll take a look through this at my next jrnl work session (next Saturday).

**Pull request merged**